### ISSUE TITLE
`register_admin_menu_item` hook not passing attributes along to template

### ISSUE THREAD

**Issue by pljspahn** [wagtail/wagtail#8087]
### Issue Summary

Adding an admin menu item via `@hooks.register('register_admin_menu_item')` does not include attributes assigned via `attrs={'my_attribute:'value'}`. I am simply trying to add `target="_blank"` to an admin menu item so it opens in a new window.

### Steps to Reproduce

1. Register admin menu item:
```
@hooks.register('register_admin_menu_item')
def register_django_admin_menu_item():
    return MenuItem('Django Admin', '/django-admin', name='django-admin', classnames='', icon_name='list-ul', attrs={'target':'_blank'}, order=500)
```

2. Inspect code in browser after page is reloaded. Adding classnames, icon_name, etc works, but attributes are not included.

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

Checking the definition [for MenuItem](https://github.com/wagtail/wagtail/blob/571b9e1918861bcbe64a9005a4dc65f3a1fe7a15/wagtail/admin/menu.py#L13) and [the assigned template](https://github.com/wagtail/wagtail/blob/571b9e1918861bcbe64a9005a4dc65f3a1fe7a15/wagtail/admin/templates/wagtailadmin/shared/menu_item.html#L6) the attributes should be included. As a quick test, I began just editing the template file but found my changes were not being shown after page reload, which says this is not the template being used or it's being cached or something. The class name assigned to the `<li>` element in the template also does not match the class name I see in browser inspector, so there's some other template being used somewhere. 

After searching for various values, I found that it seems that React is being used for the sidebar. I haven't used React much at all (I've always preferred Angular), but looking through the files for the sidebar, it seems that maybe [this definition](https://github.com/wagtail/wagtail/blob/af942a27e41b47e257b6cd46c01a13cd381fed04/client/src/components/Sidebar/menu/LinkMenuItem.tsx#L58) is responsible (I'm not entirely sure) and I can see the class does not include anything for attributes:
```
export class LinkMenuItemDefinition implements MenuItemDefinition {
  name: string;
  label: string;
  url: string;
  iconName: string | null;
  classNames?: string;
  // shouldn't attributes also be here as well as in the component template?
  ....
```

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (no)

### Technical details

* Python version: 3.9.7
* Django version: 4.0.2
* Wagtail version: 2.16
* Browser version: Firefox 97

**Comment by lb-** [comment]
@pljspahn thanks for raising this issue. 

As it stands in Wagtail 2.16 - all support for arbitrary html (this includes attributes on the link element- was dropped in adopting the new sidebar (aka main menu). 
https://docs.wagtail.org/en/stable/releases/2.16.html#deprecated-sidebar-capabilities

This also means that there is no support for any kind of html template as the DOM is all rendered in React. 

Class name is still supported though.

It would be great to get some more insights from you as to the `attrs` idea - this is similar to how Django field widgets work. Would you see the need for more than target or do you think a discrete target attribute would work?

Additionally - once we open up the target attribute we would need to be careful to enforce the rel noreferrer attribute also. 
https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/noreferrer

**Comment by pljspahn** [comment]
After I walked away from this, I had the thought that React would probably want to complain about other unexpected attributes, so that makes sense.

I think being able to add my own attributes would be handy. I also have HTMX being used in the Django admin and I could maybe see wanting to use it in Wagtail admin also, but I don't have any specific needs at the moment. I think that whatever I might want in the future, I would probably just do it in Django admin or find some other workaround. 

The workaround in this case is to just tell my users to click the link with the middle mouse button, which is simple enough.

**Comment by thibaudcolas** [comment]
Implemented in #8310! Thank you for the report @pljspahn :)
